airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeS3StagingDestination.java

```diff
@@ -57,27 +56,13 @@ public AirbyteConnectionStatus check(final JsonNode config) {
     }
     final NamingConventionTransformer nameTransformer = getNamingResolver();
     final SnowflakeS3StagingSqlOperations snowflakeS3StagingSqlOperations =
-        new SnowflakeS3StagingSqlOperations(nameTransformer, s3Config.getS3Client(), s3Config, encryptionConfig);
-    final DataSource dataSource = getDataSource(config);
-    try {
-      final JdbcDatabase database = getDatabase(dataSource);
-      final String outputSchema = super.getNamingResolver().getIdentifier(config.get("schema").asText());
-      attemptTableOperations(outputSchema, database, nameTransformer, snowflakeS3StagingSqlOperations,
-          true);
+      new SnowflakeS3StagingSqlOperations(nameTransformer, s3Config.getS3Client(), s3Config, encryptionConfig);
+    final JdbcDatabase database = getDatabase(dataSource);
+    for (final String outputSchema : getOutputSchemas(config, catalog)) {
+      attemptTableOperations(outputSchema, database, nameTransformer, snowflakeS3StagingSqlOperations, true);
       attemptStageOperations(outputSchema, database, nameTransformer, snowflakeS3StagingSqlOperations);
```

Comment:
how do you feel about defining an `attemptStageOperations(...)` method in abstractjdbcdestination? then instead of copypasting the entire checkConnectionStatus method, we could override just that method in the snowflake/redshift subclasses